flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogTableBuilder.java

```diff
@@ -66,11 +68,23 @@
 
 	private final boolean isGeneric;
 
+	private final List<String> partitionKeys;
+
 	private Map<String, String> properties = Collections.emptyMap();
 
-	public CatalogTableBuilder(ConnectorDescriptor connectorDescriptor, TableSchema tableSchema) {
+	public CatalogTableBuilder(
+			ConnectorDescriptor connectorDescriptor,
+			TableSchema tableSchema) {
+		this(connectorDescriptor, tableSchema, new ArrayList<>());
+	}
+
+	public CatalogTableBuilder(
+			ConnectorDescriptor connectorDescriptor,
+			TableSchema tableSchema,
+			List<String> partitionKeys) {
 		super(connectorDescriptor);
 		this.tableSchema = Preconditions.checkNotNull(tableSchema);
+		this.partitionKeys = Preconditions.checkNotNull(partitionKeys);
```

Comment:
since it's optional, shall the `partitionKeys` be off the constructor and loaded via "withPartitionKeys(...)"? otherwise it doesn't make sense to have builder pattern